src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/AggregatorStore.cs

```diff
@@ -0,0 +1,472 @@
+                {
+                    return multiSizeState.GetFixedSizeLabelNameDictionary<TStringSequence, TObjectSequence>().GetValuesDictionary(names);
+                }
+                else
+                {
+                    MultiSizeLabelNameDictionary<TAggregator> newState = new(state);
+                    Interlocked.CompareExchange(ref _stateUnion, newState, state);
+                    continue;
+                }
+            }
+        }
+    }
+
+    internal class MultiSizeLabelNameDictionary<TAggregator> where TAggregator : Aggregator
+    {
+        private TAggregator? NoLabelAggregator;
+        private FixedSizeLabelNameDictionary<StringSequence1, ObjectSequence1, TAggregator>? Label1;
+        private FixedSizeLabelNameDictionary<StringSequence2, ObjectSequence2, TAggregator>? Label2;
+        private FixedSizeLabelNameDictionary<StringSequence3, ObjectSequence3, TAggregator>? Label3;
+        private FixedSizeLabelNameDictionary<StringSequenceMany, ObjectSequenceMany, TAggregator>? LabelMany;
+
+        public MultiSizeLabelNameDictionary(object initialLabelNameDict)
+        {
+            NoLabelAggregator = default;
+            Label1 = default;
+            Label2 = default;
+            Label3 = default;
+            LabelMany = default;
+            if (initialLabelNameDict is TAggregator val0)
+            {
+                NoLabelAggregator = val0;
+            }
+            else if (initialLabelNameDict is FixedSizeLabelNameDictionary<StringSequence1, ObjectSequence1, TAggregator> val1)
+            {
+                Label1 = val1;
+            }
+            else if (initialLabelNameDict is FixedSizeLabelNameDictionary<StringSequence2, ObjectSequence2, TAggregator> val2)
+            {
+                Label2 = val2;
+            }
+            else if (initialLabelNameDict is FixedSizeLabelNameDictionary<StringSequence3, ObjectSequence3, TAggregator> val3)
+            {
+                Label3 = val3;
+            }
+            else if (initialLabelNameDict is FixedSizeLabelNameDictionary<StringSequenceMany, ObjectSequenceMany, TAggregator> valMany)
+            {
+                LabelMany = valMany;
+            }
+        }
+
+        public TAggregator GetNoLabelAggregator(Func<TAggregator> createFunc)
+        {
+            if (NoLabelAggregator == null)
+            {
+                Interlocked.CompareExchange(ref NoLabelAggregator, createFunc(), null);
+            }
+            return NoLabelAggregator;
+        }
+
+        public FixedSizeLabelNameDictionary<TStringSequence, TObjectSequence, TAggregator> GetFixedSizeLabelNameDictionary<TStringSequence, TObjectSequence>()
+            where TStringSequence : IStringSequence, IEquatable<TStringSequence>
+            where TObjectSequence : IObjectSequence, IEquatable<TObjectSequence>
+        {
+            TStringSequence? seq = default;
+            if (seq is StringSequence1)
+            {
+                if (Label1 == null)
```

Comment:
Some of this code is using default, some is using null.  It'd be good to be consistent.  Personally I'd prefer if all of the reference types used null.

Reply:
I swapped default to null where allowable